### PR TITLE
Misc Android fixes, add gradle flavor for gltfio-lite.

### PR DESCRIPTION
**Pull request by prideout**
This adds a maven package and fixes some Android issues, including:

- Reduce `filament-utils` from 230 KB to 190 KB.

- Some samples failed to build due to the "more than one file" gradle error. Fixed this by adopting the flavor strategy used by filamat-lite.

- Some of the samples that use the new utils library failed at run time, because the NioUtils Java class wasn't found.  Fixed this by ensuring that Filament is loaded first.

### CODE REVIEW

File: android/filament-utils-android/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,16 @@ include_directories(${FILAMENT_DIR}/include
         ..
         ../../libs/utils/include)
 
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -fno-stack-protector")
```
Comment (romainguy) [inline android/filament-utils-android/CMakeLists.txt]: This reminds me we should really switch all of our CMake files to the proper way of setting compiler flags. This is the old hacky way.

File: android/gltfio-android/gradle.properties [hunk 2]
```diff
@@ -1,3 +1,4 @@
 POM_NAME=Filament glTF Loader
-POM_ARTIFACT_ID=gltfio-android
+POM_ARTIFACT_ID_FULL=gltfio-android-full
```
Comment (romainguy) [inline android/gltfio-android/gradle.properties]: This changing the name of the Maven artifact. Can you update the root README?

File: android/filament-utils-android/CMakeLists.txt (hunk 1 above)
Comment (prideout) [reply android/filament-utils-android/CMakeLists.txt]: Agreed, we should do a clean up pass at some point.

File: android/gltfio-android/gradle.properties (hunk 2 above)
Comment (prideout) [reply android/gltfio-android/gradle.properties]: After thinking about this, I think it's better to keep the old name.  It's less disruptive to current users (like Tim) and the full version is still < 1 MB so IMO it should still considered the "default" flavor.  The root README now mentions the lite flavor.

**Pull request merged**